bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/CommandHandler.java

```diff
@@ -0,0 +1,912 @@
+        } catch (NumberFormatException e) {
+            logger.warn("malformed int parameter in for item {}", c.getChannelName());
+        }
+        return iv;
+    }
+
+    @NonNullByDefault
+    public static class WarnCommandHandler extends CommandHandler {
+        WarnCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            logger.warn("{}: command {} is not implemented yet!", nm(), cmd);
+        }
+    }
+
+    @NonNullByDefault
+    public static class NoOpCommandHandler extends CommandHandler {
+        NoOpCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            // do nothing, not even log
+        }
+    }
+
+    @NonNullByDefault
+    public static class LightOnOffCommandHandler extends CommandHandler {
+        LightOnOffCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                int ext = getIntParameter("ext", 0);
+                int direc = 0x00;
+                int level = 0x00;
+                Msg m = null;
+                if (cmd == OnOffType.ON) {
+                    level = getMaxLightLevel(conf, 0xff);
+                    direc = 0x11;
+                    logger.debug("{}: sent msg to switch {} to {}", nm(), dev.getAddress(),
+                            level == 0xff ? "on" : level);
+                } else if (cmd == OnOffType.OFF) {
+                    direc = 0x13;
+                    logger.debug("{}: sent msg to switch {} off", nm(), dev.getAddress());
+                }
+                if (ext == 1 || ext == 2) {
+                    byte[] data = new byte[] { (byte) getIntParameter("d1", 0), (byte) getIntParameter("d2", 0),
+                            (byte) getIntParameter("d3", 0) };
+                    m = dev.makeExtendedMessage((byte) 0x0f, (byte) direc, (byte) level, data);
+                    logger.debug("{}: was an extended message for device {}", nm(), dev.getAddress());
+                    if (ext == 1) {
+                        m.setCRC();
+                    } else if (ext == 2) {
+                        m.setCRC2();
+                    }
+                } else {
+                    m = dev.makeStandardMessage((byte) 0x0f, (byte) direc, (byte) level, getGroup(conf));
+                }
+                logger.debug("Sending message to {}", dev.getAddress());
+                dev.enqueueMessage(m, feature);
+                // expect to get a direct ack after this!
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
```

Comment:
This is actually here because Msg.makeMessage() throws an IOException if it can't find the message. This should never happen, except during development. Can I leave as is?

Reply:
It's at https://github.com/robnielsen/openhab-addons/blob/insteon/bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/message/Msg.java#L590. Maybe it should throw a different exception.

Reply:
Ok, thanks. Yes, an `IllegalArgumentException` might be better for this situation.
But with that, I agree that this case should not happen and if it does, the user should report it as a bug, so a warn log is fine.

Reply:
done

Reply:
I thought about using IllegalArgumentException, but it is a unchecked exception. Any other suggestions? I want to make sure the exception is caught.

Reply:
You could create a custom checked exception. Call it InvalidResponseException or some such.

Reply:
Yes, that is what I was thinking but was checking first to see if there was a different one I could use.

Reply:
done, created custom exception to replace IOException and left as warn.